docker-app/qfieldcloud/core/models.py

```diff
@@ -740,6 +740,25 @@ class Meta:
 
     is_public = models.BooleanField(default=False)
 
+    created_by = models.ForeignKey(
+        User,
+        on_delete=models.SET_NULL,
+        related_name="created_organization_members",
+        null=True,
+        blank=True,
+        limit_choices_to=models.Q(type=User.Type.PERSON),
+    )
+    created_at = models.DateTimeField(auto_now_add=True)
+    updated_by = models.ForeignKey(
+        User,
+        on_delete=models.SET_NULL,
```

Comment:
What we can do is when the user is being deleted, the fields are updated to point to the project owner (if it is a Person) or the organization owner (if it is an Organization). We can achieve that with signals.

Reply:
into the signals.py to implement something like this:

@receiver(post_delete, sender=User)
def update_organization_member_fields(sender, instance, **kwargs):
     pass
     
 ?

Reply:
yes, correct. try this one

Reply:
it seems to fail as the ownership is not changing. The code for the signals.py:

`@receiver(pre_delete, sender=User)
def update_project_owner_on_member_delete(sender, instance, **kwargs):
    """
    Update project ownership when an organization member is deleted.
    """

    fallback_user = None

    if hasattr(instance, 'organization') and instance.organization:
        fallback_user = instance.organization.organization_owner
    elif hasattr(instance, 'project') and instance.project.owner.type == User.Type.PERSON:
        fallback_user = instance.project.owner

    if fallback_user:
        OrganizationMember.objects.filter(created_by=instance).update(created_by=fallback_user)
        OrganizationMember.objects.filter(updated_by=instance).update(updated_by=fallback_user)`

Reply:
Fixed !